heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java

```diff
@@ -431,16 +433,14 @@ public boolean submitTopology() {
     return isSuccessful;
   }
 
-  protected boolean validateSubmit(SchedulerStateManagerAdaptor adaptor, String topologyName) {
+  protected void validateSubmit(SchedulerStateManagerAdaptor adaptor, String topologyName)
+      throws TopologySubmissionException {
     // Check whether the topology has already been running
     Boolean isTopologyRunning = adaptor.isTopologyRunning(topologyName);
 
     if (isTopologyRunning != null && isTopologyRunning.equals(Boolean.TRUE)) {
-      LOG.severe("Topology already exists");
-      return false;
+      throw new TopologySubmissionException("Topology already exists");
```

Comment:
We should include the topology name in the message as well:

Topology foo already exists.